front/lib/api/assistant/globalAgents.ts

```diff
@@ -0,0 +1,128 @@
+import { Authenticator } from "@app/lib/auth";
+import { DataSource } from "@app/lib/models";
+import { AgentConfigurationType } from "@app/types/assistant/agent";
+
+/**
+ * GLOBAL AGENTS CONFIGURATION
+ *
+ * To add an agent:
+ * - Add a unique SID in GLOBAL_AGENTS_SID.
+ * - Add a unique ID in GLOBAL_AGENTS_ID.
+ * - Add a case in getGlobalAgent with associated function.
+ */
+
+enum GLOBAL_AGENTS_SID {
+  GPT4 = "6b5203c158",
+  Slack = "4c974a7ac9",
+}
+enum GLOBAL_AGENTS_ID {
+  GPT4 = -1,
+  Slack = -2,
+}
+
+async function _getGPT4GlobalAgent(): Promise<AgentConfigurationType> {
+  return {
+    id: GLOBAL_AGENTS_ID.GPT4,
+    sId: GLOBAL_AGENTS_SID.GPT4,
+    name: "GPT4",
+    pictureUrl: null,
+    status: "active",
+    scope: "global",
+    generation: {
+      id: 0,
+      prompt: "Answer the following question:",
+      model: {
+        providerId: "openai",
+        modelId: "gpt-4-0613",
+      },
+    },
+    action: null,
+  };
+}
+
+async function _getSlackGlobalAgent(
+  auth: Authenticator
+): Promise<AgentConfigurationType> {
+  const workspace = auth.workspace();
+  const workspaceId = auth.workspace()?.id;
+  if (!workspace || !workspaceId) {
+    throw new Error("Cannot find Global Agent Configuration: no workspace.");
+  }
+
+  const slackDataSources = await DataSource.findAll({
```

Comment:
Should error if none?

Reply:
I returned null so that `_getSlackGlobalAgent` holds the logic of whether or not it can be activated. Works for you?

Reply:
Tots